File: core/bindings/java/com/mpiannucci/buoyfinder/core/BuoyStationItemViewData.kt
package com.mpiannucci.buoyfinder.core

class BuoyStationItemViewData {

    val ptr: Long

    constructor(rawPtr: Long) {
        ptr = rawPtr
    }

    fun finalize() {
        free(ptr)
    }

    val title: String
        get() = title(ptr)

    val subtitle: String
        get() = subtitle(ptr)

    val onClickId: String
        get() = onClickId(ptr)

    val icon: BuoyStationIcon
        get() = BuoyStationIcon.fromInt(icon(ptr))

    val color: Color
        get() = color(ptr)

    val location: Location 
        get() = Location(location(ptr))

    private external fun free(ptr: Long)
    private external fun title(ptr: Long): String
    private external fun subtitle(ptr: Long): String
    private external fun onClickId(ptr: Long): String
    private external fun icon(ptr: Long): Int
    private external fun color(ptr: Long): Color
    private external fun location(ptr: Long): Long
}

File: core/bindings/swift/BuoyStationItemViewData.swift
//
//  BuoyStationItemView.swift
//  BuoyFinder
//
//  Created by Matthew Iannucci on 9/20/18.
//  Copyright © 2018 Matthew Iannucci. All rights reserved.
//

import Foundation

class BuoyStationItemViewData {
    
    public let raw: OpaquePointer
    
    init(ptr: OpaquePointer) {
        raw = ptr
    }
    
    deinit {
        buoy_station_item_view_data_free(raw)
    }

    public var title: String {
        get {
            return String(cString: buoy_station_item_view_data_title(raw))
        }
    }
    
    public var subtitle: String {
        get {
            return String(cString: buoy_station_item_view_data_subtitle(raw))
        }
    }
    
    public var onClickId: String {
        get {
            return String(cString: buoy_station_item_view_data_on_click_id(raw))
        }
    }
    
    public var icon: BuoyStationIcon {
        get {
            return buoy_station_item_view_data_icon(raw)
        }
    }
    
    public var color: Color {
        get {
            return buoy_station_item_view_data_color(raw)
        }
    }

    public var location: Location {
        get {
            return Location(ptr: buoy_station_item_view_data_location(raw))
        }
    }
}
